portfolio/src/main/webapp/script.js

```diff
@@ -12,6 +12,37 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+/** Uses fetch() to add a message to the DOM. */
+ function getMessage() {
+    // Fetches from the DataServlet.
+    const responsePromise = fetch("/data");
```

Comment:
You may want to put "/data" in a constant (MESSAGE_API_URL or something like that). This way, if you ever rename this endpoint, you only have to update the path in one location.